tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,183 @@
+
+      const expected_sp_meta_url=`${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`;
+      const find_results_sp_meta = await page.locator('span',{'hasText' : expected_sp_meta_url})
+      await expect(find_results_sp_meta).toHaveText(expected_sp_meta_url);
+      console.log(`got span with SP metadata url: ${await find_results_sp_meta.count()}`)
+
+      const sp_meta_response = await axios({
+        url: `${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`,
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+          'Content-Type': 'application/json'
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false
+        })
+      });
+
+      if (sp_meta_response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const service_provider_metadata=sp_meta_response.data.replaceAll('Signed="true"','Signed="false"')
+      fs.writeFileSync('fixtures/service_provider_metadata.xml', service_provider_metadata);
+
+      await page.goto('https://samltest.id/upload.php');
+      await page.waitForSelector('input[id=uploader]');
+
+      // Click input[name="userfile"]
+      await page.locator('input[name="userfile"]').click();
+      // Upload m
+      await page.locator('input[name="userfile"]').setInputFiles('fixtures/service_provider_metadata.xml');
+      // Click input:has-text("Upload")
+      await page.locator('input:has-text("Upload")').click();
+      await expect(page).toHaveURL('https://samltest.id/upload.php');
+
+      console.log('uploaded file, making screen shot, after waiting 4s');
+      await page.waitForTimeout(timeout_4s);
+      await page.screenshot({path: 'saml-uploaded.png'});
+
+      // Go to https://staging.hosted.mender.io/ui/settings/user-management
+      await page.goto(`${baseUrl}ui/settings/user-management`);
+      // Click text=Create new user
+      await page.locator('text=Create new user').click();
+      // Click [placeholder="Email"]
+      await page.locator('[placeholder="Email"]').click();
+      // Fill [placeholder="Email"]
+      await page.locator('[placeholder="Email"]').fill(saml_settings['email']);
+      // Click text=Create user
+      await page.locator('text=Create user').click();
+      await page.waitForTimeout(timeout_1s);
+      await expect(page.locator(`text=${saml_settings['email']}`)).toHaveText(saml_settings['email']);
+
+      console.log(`logging in via ${baseUrl}api/management/v1/useradm/auth/sso/${id}/login`);
+      // enter valid username and password
+      await page.goto(`${baseUrl}api/management/v1/useradm/auth/sso/${id}/login`);
+      await page.waitForTimeout(timeout_4s);
+      await page.screenshot({path: 'saml-redirected.png'});
+
+      // Double click text=morty
+      await page.locator('text=morty').dblclick();
+
+      // Click input[name="j_username"]
+      await page.locator('input[name="j_username"]').click();
```

Comment:
my guess would be this selector is from the samltest login, in this case we have to be extra careful with the selectors as they might change (not sure if they are generated, but their next deployment might be `k_username`), so maybe something like `[placeholder=username]` would work (if there is a placeholder displayed)

Reply:
this is the samltes.id login page, there are no placeholders there. I do not think they change it very often. I can do more elaborate search for fields, do you think it is necessary?

Reply:
what might be more important is the search for reliably stable selectors... 👌  I'll take a look to see if there might be  other selectors that might fit...

Reply:
I believe the input name is pretty stable -- if you change that you will have to change the code that handles the form on the other side. I would keep them (I could not find more reliable one)

Reply:
Yes, that is e.g. how server side rendering works - my guess would be the saml test code is in PHP, where such a change happens in a single place... But we can keep it and fix if it breaks...

Reply:
roger that. and I know it is in php :)